File: src/data/coords.ts
// Coordonnées GPS approximatives des villes et communes de Côte d'Ivoire.
// Utilisées comme position par défaut d'une annonce quand le vendeur n'active
// pas son GPS, afin de calculer une distance indicative.

export interface Coords {
  lat: number
  lng: number
}

// Communes du District Autonome d'Abidjan
export const communeCoords: Record<string, Coords> = {
  Abobo: { lat: 5.4165, lng: -4.0159 },
  Adjamé: { lat: 5.3646, lng: -4.0227 },
  Anyama: { lat: 5.4949, lng: -4.0517 },
  Attécoubé: { lat: 5.3333, lng: -4.0333 },
  Bingerville: { lat: 5.3556, lng: -3.8869 },
  Cocody: { lat: 5.3599, lng: -3.9877 },
  Koumassi: { lat: 5.2953, lng: -3.9469 },
  Marcory: { lat: 5.3, lng: -3.9833 },
  'Le Plateau': { lat: 5.3242, lng: -4.0189 },
  'Port-Bouët': { lat: 5.2558, lng: -3.9269 },
  Songon: { lat: 5.2917, lng: -4.2528 },
  Treichville: { lat: 5.2917, lng: -4.0083 },
  Yopougon: { lat: 5.3456, lng: -4.0708 },
}

// Villes (par id de ville)
export const cityCoords: Record<string, Coords> = {
  'abidjan-ville': { lat: 5.3599, lng: -4.0083 },
  'yamoussoukro-ville': { lat: 6.8276, lng: -5.2893 },
  bouake: { lat: 7.6939, lng: -5.0303 },
  daloa: { lat: 6.8776, lng: -6.4502 },
  'san-pedro-ville': { lat: 4.7485, lng: -6.6363 },
  korhogo: { lat: 9.4578, lng: -5.6294 },
  man: { lat: 7.4125, lng: -7.5538 },
  gagnoa: { lat: 6.1319, lng: -5.9506 },
  divo: { lat: 5.8397, lng: -5.3572 },
  abengourou: { lat: 6.7297, lng: -3.4964 },
  'grand-bassam': { lat: 5.2118, lng: -3.7386 },
  bonoua: { lat: 5.2725, lng: -3.5964 },
  aboisso: { lat: 5.4676, lng: -3.2072 },
  agboville: { lat: 5.9282, lng: -4.2131 },
  dabou: { lat: 5.3256, lng: -4.3767 },
  adzope: { lat: 6.1069, lng: -3.8619 },
  seguela: { lat: 7.9611, lng: -6.6731 },
  odienne: { lat: 9.5, lng: -7.5667 },
  bondoukou: { lat: 8.0402, lng: -2.8, },
  ferkessedougou: { lat: 9.5928, lng: -5.1944 },
  soubre: { lat: 5.7836, lng: -6.5936 },
  toumodi: { lat: 6.5528, lng: -5.0189 },
  katiola: { lat: 8.1333, lng: -5.1},
  bouafle: { lat: 6.99, lng: -5.7444 },
  issia: { lat: 6.4922, lng: -6.5872 },
  duekoue: { lat: 6.7419, lng: -7.3436 },
  guiglo: { lat: 6.5439, lng: -7.4869 },
  daoukro: { lat: 7.0586, lng: -3.9631 },
  dimbokro: { lat: 6.6486, lng: -4.7061 },
  boundiali: { lat: 9.5217, lng: -6.4869 },
  mankono: { lat: 8.0586, lng: -6.1897 },
  touba: { lat: 8.2833, lng: -7.6833 },
  bouna: { lat: 9.2667, lng: -3.0 },
  tabou: { lat: 4.4231, lng: -7.3528 },
  danane: { lat: 7.2597, lng: -8.155 },
  sinfra: { lat: 6.6197, lng: -5.9161 },
  lakota: { lat: 5.8481, lng: -5.6786 },
  bongouanou: { lat: 6.6511, lng: -4.2039 },
  minignan: { lat: 10.0, lng: -7.7333 },
  'grand-lahou': { lat: 5.1386, lng: -5.0203 },
  sassandra: { lat: 4.9503, lng: -6.0836 },
  fresco: { lat: 5.0939, lng: -5.5658 },
  meagui: { lat: 5.8833, lng: -6.5333 },
}

/** Retourne les coordonnées approximatives d'une commune, sinon d'une ville. */
export function coordsFor(cityId?: string, commune?: string): Coords | undefined {
  if (commune && communeCoords[commune]) return communeCoords[commune]
  if (cityId && cityCoords[cityId]) return cityCoords[cityId]
  return undefined
}

File: flutter_app/lib/data/coords.dart
// =============================================================================
//  Coordonnées GPS approximatives des villes et communes de Côte d'Ivoire —
//  port fidèle de src/data/coords.ts.
//
//  Servent de position par défaut d'une annonce quand le vendeur n'active pas
//  son GPS, afin de calculer une distance indicative sur la fiche.
// =============================================================================
library;

/// Un point GPS.
class Coords {
  final double lat;
  final double lng;
  const Coords(this.lat, this.lng);
}

/// Communes du District Autonome d'Abidjan.
const Map<String, Coords> communeCoords = {
  'Abobo': Coords(5.4165, -4.0159),
  'Adjamé': Coords(5.3646, -4.0227),
  'Anyama': Coords(5.4949, -4.0517),
  'Attécoubé': Coords(5.3333, -4.0333),
  'Bingerville': Coords(5.3556, -3.8869),
  'Cocody': Coords(5.3599, -3.9877),
  'Koumassi': Coords(5.2953, -3.9469),
  'Marcory': Coords(5.3, -3.9833),
  'Le Plateau': Coords(5.3242, -4.0189),
  'Port-Bouët': Coords(5.2558, -3.9269),
  'Songon': Coords(5.2917, -4.2528),
  'Treichville': Coords(5.2917, -4.0083),
  'Yopougon': Coords(5.3456, -4.0708),
};

/// Villes (par identifiant de ville).
const Map<String, Coords> cityCoords = {
  'abidjan-ville': Coords(5.3599, -4.0083),
  'yamoussoukro-ville': Coords(6.8276, -5.2893),
  'bouake': Coords(7.6939, -5.0303),
  'daloa': Coords(6.8776, -6.4502),
  'san-pedro-ville': Coords(4.7485, -6.6363),
  'korhogo': Coords(9.4578, -5.6294),
  'man': Coords(7.4125, -7.5538),
  'gagnoa': Coords(6.1319, -5.9506),
  'divo': Coords(5.8397, -5.3572),
  'abengourou': Coords(6.7297, -3.4964),
  'grand-bassam': Coords(5.2118, -3.7386),
  'bonoua': Coords(5.2725, -3.5964),
  'aboisso': Coords(5.4676, -3.2072),
  'agboville': Coords(5.9282, -4.2131),
  'dabou': Coords(5.3256, -4.3767),
  'adzope': Coords(6.1069, -3.8619),
  'seguela': Coords(7.9611, -6.6731),
  'odienne': Coords(9.5, -7.5667),
  'bondoukou': Coords(8.0402, -2.8),
  'ferkessedougou': Coords(9.5928, -5.1944),
  'soubre': Coords(5.7836, -6.5936),
  'toumodi': Coords(6.5528, -5.0189),
  'katiola': Coords(8.1333, -5.1),
  'bouafle': Coords(6.99, -5.7444),
  'issia': Coords(6.4922, -6.5872),
  'duekoue': Coords(6.7419, -7.3436),
  'guiglo': Coords(6.5439, -7.4869),
  'daoukro': Coords(7.0586, -3.9631),
  'dimbokro': Coords(6.6486, -4.7061),
  'boundiali': Coords(9.5217, -6.4869),
  'mankono': Coords(8.0586, -6.1897),
  'touba': Coords(8.2833, -7.6833),
  'bouna': Coords(9.2667, -3.0),
  'tabou': Coords(4.4231, -7.3528),
  'danane': Coords(7.2597, -8.155),
  'sinfra': Coords(6.6197, -5.9161),
  'lakota': Coords(5.8481, -5.6786),
  'bongouanou': Coords(6.6511, -4.2039),
  'minignan': Coords(10.0, -7.7333),
  'grand-lahou': Coords(5.1386, -5.0203),
  'sassandra': Coords(4.9503, -6.0836),
  'fresco': Coords(5.0939, -5.5658),
  'meagui': Coords(5.8833, -6.5333),
};

/// Les coordonnées approximatives d'une commune, sinon d'une ville, sinon null.
Coords? coordsFor(String? cityId, [String? commune]) {
  if (commune != null && communeCoords.containsKey(commune)) {
    return communeCoords[commune];
  }
  if (cityId != null && cityCoords.containsKey(cityId)) {
    return cityCoords[cityId];
  }
  return null;
}
